server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/ElasticsearchUtil.java

```diff
@@ -47,11 +49,14 @@
      */
     public static BoolQueryBuilder buildQuery(Criteria criteria, String timeProperty,
             String businessTxnProperty, Class<?> targetClass) {
-        long startTime = criteria.calculateStartTime();
-        long endTime = criteria.calculateEndTime();
-
+        /**
+         * Internally all time units are stored in microseconds
+         * Criteria API accepts milliseconds, therefore range adjustment is needed
+         */
         BoolQueryBuilder query = QueryBuilders.boolQuery()
-                .must(QueryBuilders.rangeQuery(timeProperty).from(startTime).to(endTime));
+                .must(QueryBuilders.rangeQuery(timeProperty)
+                        .from(TimeUnit.MILLISECONDS.toMicros(criteria.calculateStartTime()))
```

Comment:
We need to check about this - if the timestamp being stored with records in the ES db is using its own timestamp (as we don't specify the mapping to our field), then it would be in milliseconds I guess - so the query should take milliseconds?
The docs aren't clear, so may need to experiment.